src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -506,6 +519,78 @@ export const createForm = (
     formParams,
   }
 
+  if (formParams.responseMode === FormResponseMode.Multirespondent) {
+    const workflow = (formParams as MultirespondentFormToCreate).workflow ?? []
+
+    const emailFieldIds = formParams.form_fields
+      ?.filter((field) => field.fieldType === BasicField.Email)
+      .map((field) => field._id.toString())
+    const isRespondentFieldEmail = workflow?.every((step) => {
+      return (
+        step.workflow_type !== WorkflowType.Dynamic ||
+        (step.field && emailFieldIds?.includes(step.field))
+      )
+    })
+    if (!isRespondentFieldEmail) {
+      return errAsync(
+        new MalformedParametersError(
+          'All respondent fields in workflow must be email fields',
+        ),
+      )
+    }
+
+    const isFirstStepApproval = workflow[0] && workflow[0].approval_field
+    if (isFirstStepApproval) {
+      return errAsync(
+        new MalformedParametersError(
+          'First step of workflow cannot be an approval step',
+        ),
+      )
+    }
+
+    const yesNoFieldIds = formParams.form_fields
+      ?.filter((field) => field.fieldType === BasicField.YesNo)
+      .map((field) => field._id.toString())
+    const isApprovalFieldYesNo = workflow.every((step) => {
+      return (
+        !step.approval_field || yesNoFieldIds?.includes(step.approval_field)
+      )
+    })
+    if (!isApprovalFieldYesNo) {
+      return errAsync(
+        new MalformedParametersError(
+          'All approval fields must be yes/no fields',
+        ),
+      )
+    }
+
+    const selectedApprovalFields = workflow
+      .map((step) => step.approval_field)
+      .filter(Boolean)
+    const isApprovalFieldUnique =
+      new Set(selectedApprovalFields).size === selectedApprovalFields.length
+    if (!isApprovalFieldUnique) {
+      return errAsync(
+        new MalformedParametersError(
+          'Each yes/no field cannot be used in more than one approval step',
+        ),
+      )
+    }
+
+    const isApprovalFieldInEditFields = workflow.every(
+      (step) =>
+        !step.approval_field ||
+        (step.edit && step.edit.includes(step.approval_field)),
+    )
+    if (!isApprovalFieldInEditFields) {
+      return errAsync(
+        new MalformedParametersError(
+          "For all steps, the selected approval field must also be included in same step's edit fields",
```

Comment:
/nit this message sounds very technical. Suggestion: "Approval fields must be included in edit fields." since admins must save on every workflow step. The context of which step can likely be omitted without loss of specificity.